src/gromo/modules/conv2d_growing_module.py

```diff
@@ -14,11 +14,382 @@
+            else:
+                return self.activity
+
+    def set_next_modules(
+        self, next_modules: list[MergeGrowingModule | GrowingModule]
+    ) -> None:
+        if self.tensor_s is not None and self.tensor_s.samples > 0:
+            warn(
+                f"You are setting the next modules of {self.name} with a non-empty tensor S."
+            )
+        self.next_modules = next_modules if next_modules else []
+        for module in self.next_modules:
+            if isinstance(module, (Conv2dGrowingModule, Conv2dMergeGrowingModule)):
+                assert module.in_channels == self.out_channels
+                module.input_size = self.output_size
+            elif isinstance(module, (LinearGrowingModule, LinearMergeGrowingModule)):
+                assert module.in_features == self.out_features
+            else:
+                raise NotImplementedError(
+                    "The next modules must be either Linear or Convolution."
+                )
+        # assert all(
+        #     modules.in_features == self.out_features for modules in self.next_modules
+        # ), f"The output features must match the input features of the next modules."
+
+    def set_previous_modules(
+        self, previous_modules: list[MergeGrowingModule | GrowingModule]
+    ) -> None:
+        if self.previous_tensor_s is not None and self.previous_tensor_s.samples > 0:
+            warn(
+                f"You are setting the previous modules of {self.name} with a non-empty previous tensor S."
+            )
+        if self.previous_tensor_m is not None and self.previous_tensor_m.samples > 0:
+            warn(
+                f"You are setting the previous modules of {self.name} with a non-empty previous tensor M."
+            )
+
+        self.previous_modules = previous_modules if previous_modules else []
+        self.total_in_features = 0
+        self.total_out_features = 0
+        for module in self.previous_modules:
+            if not isinstance(module, (Conv2dGrowingModule, Conv2dMergeGrowingModule)):
+                raise TypeError(
+                    "The previous modules must be Conv2dGrowingModule or Conv2dMergeGrowingModule."
+                )
+            # if module.out_features != self.in_features:
+            #     raise ValueError(
+            #         "The input features must match the output features of the previous modules."
+            #     )
+            if module.out_channels != self.in_channels:
+                raise ValueError(
+                    "The input channels must match the output channels of the previous modules."
+                )
+            self.total_in_features += module.in_features + module.use_bias
+            self.total_out_features += module.out_features
+            # self.total_out_features += module.use_bias
+        if self.total_in_features > 0:
+            if self.input_size is None:
+                self.input_size = (module.out_width, module.out_height)
```

Comment:
We should not use `module` outside of the for loop. This is not supposed to happen.
```suggestion
                self.input_size = (self.previous_module[0].out_width, self.previous_module[0].out_height)
```